File: lib/go/internal/mock/IsSubMap.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package mock

// IsSubMap verifies that every entry in part matches the corresponding entry in whole.
func IsSubMap(part map[string]any, whole map[string]any) bool {
	if len(part) == 0 {
		return true
	}

	for key, partValue := range part {
		wholeValue, ok := whole[key]
		if !ok {
			return false
		}

		if !IsSubMapEntryEqual(partValue, wholeValue) {
			return false
		}
	}

	return true
}

File: lib/ts/src/internal/mock/IsSubMap.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { isSubMapEntryEqual } from '../../internal/mock/IsSubMapEntryEqual.js';

export function isSubMap(part: Record<string, any>, whole: Record<string, any>): boolean {
    for (const partKey in part) {
        const wholeValue = whole[partKey] as Record<string, any>;
        const partValue = part[partKey] as Record<string, any>;
        const entryIsEqual = isSubMapEntryEqual(partValue, wholeValue);
        if (!entryIsEqual) {
            return false;
        }
    }
    return true;
}
